### PR TITLE
fix: Schedule usage reports as individual tasks

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

We don't seem to get all usage reports for every org and my theory is that we are trying to do too much in a single tasks.

## Changes

* Breaks out the usage reporting into single tasks with an overall task for scheduling for all known orgs
* Adds helper method for awaiting tasks this way

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Tests

## TODO:

- [x] Fix send_usage_report command

### CODE REVIEW

File: ee/tasks/subscriptions/subscription_utils.py [hunk 1]
```diff
@@ -41,10 +42,6 @@ def generate_assets(
     tasks = [exporter.export_asset.s(asset.id) for asset in assets]
     parallel_job = group(tasks).apply_async()
 
-    start_time = datetime.now()
-    while not parallel_job.ready():
-        sleep(1)
-        if datetime.now() > start_time + ASSET_GENERATION_MAX_TIMEOUT:
-            raise Exception("Timed out waiting for exports")
+    wait_for_parallel_celery_group(parallel_job, max_timeout=ASSET_GENERATION_MAX_TIMEOUT)
```
Comment (kappa90) [inline ee/tasks/subscriptions/subscription_utils.py:45]: This has nothing to do with the usage report?

File: ee/tasks/usage_report.py [hunk 2]
```diff
@@ -187,7 +188,7 @@ def get_org_usage_report(period: Tuple[datetime, datetime], team_ids: List[str])
             multiple_ids_per_person=count_total_persons_with_multiple_ids(team_id),
             group_types_total=GroupTypeMapping.objects.filter(team_id=team_id).count(),
```
Comment (kappa90) [inline ee/tasks/usage_report.py:189]: What about taking all of these out and run them before the dataclass is created, as we said earlier?

File: ee/tasks/usage_report.py [hunk 3]
```diff
@@ -430,3 +355,65 @@ def fetch_sql(sql_: str, params: Tuple[Any, ...]) -> List[Any]:
...
+    try:
+        org_owner = get_org_owner_or_first_user(organization_id)
+
+        if not org_owner:
+            raise Exception("No owner found for organization")
+
+        logger.info("Sending usage report for organization %s ...", organization_id)
+
+        distinct_id = org_owner.distinct_id
+        usage = get_org_usage_report(period, team_ids)
```
Comment (kappa90) [inline ee/tasks/usage_report.py:385]: We should have a minimum set of usage data (events, recordings counts) and just send these if anything else is failing

File: ee/tasks/subscriptions/subscription_utils.py (hunk 1 above)
Comment (benjackwhite) [reply ee/tasks/subscriptions/subscription_utils.py:45]: Yeah it's just that I took out the code for waiting for parallel tasks into a util function

File: ee/tasks/usage_report.py (hunk 3 above)
Comment (benjackwhite) [reply ee/tasks/usage_report.py:385]: Agreed but going to do that as a separate PR as it needs some thinking...

**Review by kappa90 — APPROVED** [review]

**Pull request merged**